Formula/opentsdb.rb

```diff
@@ -0,0 +1,147 @@
+    end
+
+    env = {
+      :HBASE_HOME => Formula["hbase"].opt_libexec,
+      :COMPRESSION => (build.with?("lzo") ? "LZO" : "NONE"),
+    }
+    env = Language::Java.java_home_env.merge(env)
+    (bin/"create_table.sh").write_env_script opt_pkgshare/"tools/create_table.sh", env
+
+    inreplace pkgshare/"etc/opentsdb/opentsdb.conf", "/usr/share", "#{HOMEBREW_PREFIX}/share"
+    etc.install Dir["#{pkgshare}/etc/opentsdb"]
+    (pkgshare/"plugins/.keep").write ""
+
+    (bin/"start-tsdb.sh").write <<-EOS.undent
+      #!/bin/sh
+      exec "#{opt_bin}/tsdb" tsd \\
+        --config="#{etc}/opentsdb/opentsdb.conf" \\
+        --staticroot="#{opt_pkgshare}/static/" \\
+        --cachedir="#{var}/cache/opentsdb" \\
+        --port=4242 \\
+        --zkquorum=localhost:2181 \\
+        --zkbasedir=/hbase \\
+        --auto-metric \\
+        "$@"
+    EOS
+  end
+
+  def post_install
+    (var/"cache/opentsdb").mkpath
+
+    system "#{Formula["hbase"].opt_libexec}/bin/start-hbase.sh"
+    sleep 2
+    system "#{opt_bin}/create_table.sh"
+    system "#{Formula["hbase"].opt_libexec}/bin/stop-hbase.sh"
+  end
+
+  plist_options :manual => "#{HOMEBREW_PREFIX}/opt/opentsdb/bin/start-tsdb.sh"
+
+  def plist; <<-EOS.undent
+    <?xml version="1.0" encoding="UTF-8"?>
+    <!DOCTYPE plist PUBLIC "-//Apple//DTD PLIST 1.0//EN" "http://www.apple.com/DTDs/PropertyList-1.0.dtd">
+    <plist version="1.0">
+    <dict>
+      <key>KeepAlive</key>
+      <dict>
+        <key>OtherJobEnabled</key>
+        <string>#{Formula["hbase"].plist_name}</string>
+      </dict>
+      <key>Label</key>
+      <string>#{plist_name}</string>
+      <key>ProgramArguments</key>
+      <array>
+        <string>#{opt_bin}/start-tsdb.sh</string>
+      </array>
+      <key>WorkingDirectory</key>
+      <string>#{HOMEBREW_PREFIX}</string>
+      <key>StandardOutPath</key>
+      <string>#{var}/opentsdb/opentsdb.log</string>
+      <key>StandardErrorPath</key>
+      <string>#{var}/opentsdb/opentsdb.err</string>
+    </dict>
+    </plist>
+    EOS
+  end
+
+  test do
+    cp_r (Formula["hbase"].libexec/"conf"), testpath
+    inreplace (testpath/"conf/hbase-site.xml") do |s|
+      s.gsub! /(hbase.rootdir.*)\n.*/, "\\1\n<value>file://#{testpath}/hbase</value>"
+      s.gsub! /(hbase.zookeeper.property.dataDir.*)\n.*/, "\\1\n<value>#{testpath}/zookeeper</value>"
+    end
+
+    ENV["HBASE_LOG_DIR"]  = testpath/"logs"
+    ENV["HBASE_CONF_DIR"] = testpath/"conf"
+    ENV["HBASE_PID_DIR"]  = testpath/"pid"
+
+    system "#{Formula["hbase"].opt_bin}/start-hbase.sh"
+    sleep 2
+
+    system "#{opt_bin}/create_table.sh"
+
+    tsdb_err = "#{testpath}/tsdb.err"
+    tsdb_out = "#{testpath}/tsdb.out"
+    pid = fork do
+      $stderr.reopen(tsdb_err, "w")
+      $stdout.reopen(tsdb_out, "w")
+      exec("#{opt_bin}/start-tsdb.sh")
+    end
+
+    sleep 15
+
+    pipe_output("nc localhost 4242 2>&1", "put homebrew.install.test 1356998400 42.5 host=webserver01 cpu=0")
```

Comment:
```diff
-    pipe_output("nc localhost 4242 2>&1", "put homebrew.install.test 1356998400 42.5 host=webserver01 cpu=0")
+    pipe_output("nc localhost 4242 2>&1", "put homebrew.install.test 1356998400 42.5 host=webserver01 cpu=0\n")
```

The newline at the end is really important here! Otherwise the request won't be properly terminated and the following `tsdb query` will fail as can be seen in the latest failed test bot build.